src/Compressors/Gaussian.jl

```diff
@@ -0,0 +1,98 @@
+struct Gaussian <: Compressor
+    n_rows::Int64
+    n_cols::Int64
+end
+
+function Gaussian(;n_rows::Int64 = 0, n_cols::Int64 = 0)
+    # Partially construct the Gaussian datatype
+    return Gaussian(n_rows, n_cols)
+end
+
+mutable struct GaussianRecipe <: CompressorRecipe
+    n_rows::Int64
+    n_cols::Int64
+    scale::Float64
+    sketch_matrix::Matrix{Float64}
+end
+
+function complete_compressor(Gaussian_info::Gaussian, A::AbstractMatrix)
+    n_rows = Gaussian_info.n_rows
+    n_cols = Gaussian_info.n_cols
+    # Find the zero dimension and set it to be the dimension of A
+    if n_rows == 0 && n_cols == 0
+        # By default we will compress the row dimension to size 2
+        n_cols = size(A, 1)
+        n_rows = 2
+        # correct these sizes
+        initial_size = max(n_rows, n_cols)
+        sample_size = min(n_rows, n_cols)
+    elseif n_rows == 0 && n_cols > 0
+        # Assuming that if n_rows is not specified we compress column dimension
+         n_rows = size(A, 2)
+         # If the user specifies one size as nonzero that is the sample size
+         sample_size = n_cols
+         initial_size = n_rows
+    elseif n_rows > 0 && n_cols == 0
+        n_cols = size(A, 1)
+        sample_size = n_rows
+        initial_size = n_cols
+    else
+        if n_rows == size(A, 2)
+            initial_size = n_rows
+            sample_size = n_cols
+        elseif n_cols == size(A, 2)
+            initial_size = n_cols
+            sample_size = n_rows
+        else
+            @assert false "Either you inputted row or column dimension must match \\
+            the column or row dimension of the matrix."
+        end
+
+    end
```

Comment:
Dispatching off of cardinality as in SparseSign is probably more appropriate here.